platform/ios/uitest/MapViewTests.m

```diff
@@ -305,6 +305,23 @@ - (void)testZoomSet {
                                @"setting zoom should take effect");
 }
 
+- (void)testCameraDebouncing {
+    MGLMapCamera *camera = [MGLMapCamera cameraLookingAtCenterCoordinate:CLLocationCoordinate2DMake(45, -122)
+                                                            fromDistance:100
+                                                                   pitch:30
+                                                                 heading:45];
+    tester.mapView.camera = camera;
+    XCTAssertEqualObjects(tester.mapView.camera, camera);
+    
+    [tester.mapView setCamera:camera withDuration:10 animationTimingFunction:nil completionHandler:^{
+        XCTAssert(NO, @"Camera animation should not be canceled by redundantly setting the camera to the current camera.");
```

Comment:
This is the best I could come up with for a test of the redundancy check. It essentially only verifies that `cancelTransitions()` isn’t called in the case of a redundant camera change. The test in #6060 is infeasible because the completion handler is eventually called even for redundant camera changes.

I verified via breakpoints that we do short-circuit mbgl for redundant camera changes and that significant camera changes still work.